gpt_engineer/chat_to_files.py

```diff
@@ -40,3 +43,83 @@ def to_files(chat, workspace):
     files = parse_chat(chat)
     for file_name, file_content in files:
         workspace[file_name] = file_content
+
+
+def overwrite_files(chat, dbs, replace_files):
+    """
+    Replace the AI files to the older local files.
+    """
+    dbs.workspace["all_output.txt"] = chat
+
+    files = parse_chat(chat)
+    for file_name, file_content in files:
+        # Verify if the file created by the AI agent was in the input list
+        if file_name in replace_files:
+            # If the AI created a file from our input list, we replace it.
+            with open(replace_files[file_name], "w") as text_file:
+                text_file.write(file_content)
+        else:
+            # If the AI create a new file I don't know where to put it yet
+            # maybe we can think in a smarter solution for this in the future
+            # like asking the AI where to put it.
+            #
+            # by now, just add this to the workspace inside .gpteng folder
+            print(
+                f"Could not find file path for '{file_name}', creating file in workspace"
+            )
+            dbs.workspace[file_name] = file_content
+
+
+def get_code_strings(input) -> dict[str, str]:
+    """
+    Read file_list.txt and return file names and its content.
+    """
+    filesPaths = input["file_list.txt"].strip().split("\n")
+    filesDict = {}
+    for filePath in filesPaths:
+        fileData = ""
+        with open(filePath, "r") as file:
+            fileData = file.read()
+        if fileData:
+            fileName = os.path.basename(filePath).split("/")[-1]
+            filesDict[fileName] = fileData
+    return filesDict
+
+
+def format_file_to_input(fileName: str, fileContent: str) -> str:
+    """
+    Format a file string to use as input to AI agent
+    """
+    filestr = f"""
+    {fileName}
+    ```
+    {fileContent}
+    ```
+    """
+    return filestr
+
+
+def ask_for_files(input) -> dict[str, str]:
+    """
+    Display a tkinter file selection window to select context files.
+    Return a dictionary of file_name as key and file_path as value
+    """
+    root = tk.Tk()
```

Comment:
I would prefer if we can run GPT engineer via ssh, ie no GUI. To facilitate that I’d suggest we ask user if they:
1. want to launch a gui selector
2. Want to reuse the file that contains files to use, or fallback to all files if that file does not exist

Reply:
Also nice to not launch the selector every time